src/diff_tform.c

```diff
@@ -318,8 +318,10 @@ static int normalize_find_opts(
 #undef USE_DEFAULT
 
 	if (!opts->rename_limit) {
-		opts->rename_limit = git_config__get_int_force(
-			cfg, "diff.renamelimit", DEFAULT_RENAME_LIMIT);
+		if (diff->repo) {
```

Comment:
Well, see a few lines above. If `diff->repo != NULL` we return -1 if `git_repository_config__weakptr(&cfg, diff->repo) < 0`. I mainly used `if (diff->repo)` because we use the same check when retrieving the "diff.renames" config.